### PR TITLE
Added a download CSV button to the median seroprevalence by who region and age group table

### PR DESCRIPTION
**Pull request by SeanKennyNF**
One step in resolving #247.

Here's what one of the downloaded CSV files looks like: 
[median-seroprevalence-by-who-region-and-age-group.csv](https://github.com/serotracker/dashboard/files/14813063/median-seroprevalence-by-who-region-and-age-group.csv)

image

Filters have an effect on the downloaded csv as well:
[median-seroprevalence-by-who-region-and-age-group(1).csv](https://github.com/serotracker/dashboard/files/14813064/median-seroprevalence-by-who-region-and-age-group.1.csv)

image

### CODE REVIEW

File: app/pathogen/arbovirus/dashboard/(visualizations)/median-seroprevalence-by-who-region-and-age-group-table.tsx [hunk 1]
```diff
@@ -75,7 +76,7 @@ export const MedianSeroprevalenceByWhoRegionAndAgeGroupTable = () => {
                 ([whoRegion, dataPoints]) => [
                   whoRegion,
                   typedGroupBy(
-                    dataPoints,
+                    dataPoints.filter((dataPoint) => Object.values(AgeGroup).includes(dataPoint.ageGroup)),
```
Comment (SeanKennyNF) [inline app/pathogen/arbovirus/dashboard/(visualizations)/median-seroprevalence-by-who-region-and-age-group-table.tsx:79]: Some `nulls` were sneaking by, we'll just deal with them like this for now to make the table and the csv consistent :)

**Comment by SeanKennyNF** [comment]
Whoops read this wrong, 1 sec.

File: app/pathogen/arbovirus/dashboard/(visualizations)/median-seroprevalence-by-who-region-and-age-group-table.tsx [hunk 2]
```diff
@@ -27,6 +28,49 @@ enum AgeGroup {
...
+  data: any[];
+}
+
+interface GetMedianSeroprevalenceInformationFromDataOutput {
+  seroprevalenceNumber: number | undefined;
+  seroprevalencePercentageString: string;
+  backgroundColourHexCode: string;
+}
+
+const getMedianSeroprevalenceInformationFromData = (
```
Comment (SeanKennyNF) [inline app/pathogen/arbovirus/dashboard/(visualizations)/median-seroprevalence-by-who-region-and-age-group-table.tsx:41]: Moved from below into function for both the csv downloader and table to use. Something more elegant might involve them formatting their stuff together? :thinking:

File: app/pathogen/arbovirus/dashboard/(visualizations)/median-seroprevalence-by-who-region-and-age-group-table.tsx [hunk 3]
```diff
@@ -75,7 +119,9 @@ export const MedianSeroprevalenceByWhoRegionAndAgeGroupTable = () => {
                 ([whoRegion, dataPoints]) => [
                   whoRegion,
                   typedGroupBy(
-                    dataPoints,
+                    dataPoints.filter((dataPoint) =>
+                      Object.values(AgeGroup).includes(dataPoint.ageGroup)
+                    ),
```
Comment (SeanKennyNF) [inline app/pathogen/arbovirus/dashboard/(visualizations)/median-seroprevalence-by-who-region-and-age-group-table.tsx:124]: Some `null`s in our data.

File: app/pathogen/arbovirus/dashboard/(visualizations)/median-seroprevalence-by-who-region-and-age-group-table.tsx [hunk 4]
```diff
@@ -97,13 +143,44 @@ export const MedianSeroprevalenceByWhoRegionAndAgeGroupTable = () => {
...
+      ([arbovirus, dataForArbovirus]) =>
+        typedObjectEntries(dataForArbovirus).flatMap(
+          ([whoRegion, dataForWhoRegion]) => ({
+            Arbovirus: arbovirus,
+            "WHO Region": whoRegion,
+            ...Object.values(AgeGroup).map((ageGroup) => {
+              const columnName = `Median Seroprevalence (${ageGroup})`
+              const { seroprevalencePercentageString: columnValue } = getMedianSeroprevalenceInformationFromData({data: dataForWhoRegion[ageGroup]})
+              return {[columnName]: columnValue}
+            }).reduce((a, b) => ({...a, ...b}))
```
Comment (SeanKennyNF) [inline app/pathogen/arbovirus/dashboard/(visualizations)/median-seroprevalence-by-who-region-and-age-group-table.tsx:157]: How I'm doing a column for each age group.

File: app/pathogen/arbovirus/dashboard/(visualizations)/median-seroprevalence-by-who-region-and-age-group-table.tsx [hunk 5]
```diff
@@ -97,13 +143,44 @@ export const MedianSeroprevalenceByWhoRegionAndAgeGroupTable = () => {
     [currentlySelectedArbovirus, tableDatasets]
   );
 
+  const downloadCsv = useCallback(() => {
+    const dataForCsv = typedObjectEntries(tableDatasets).flatMap(
+      ([arbovirus, dataForArbovirus]) =>
+        typedObjectEntries(dataForArbovirus).flatMap(
+          ([whoRegion, dataForWhoRegion]) => ({
```
Comment (SeanKennyNF) [inline app/pathogen/arbovirus/dashboard/(visualizations)/median-seroprevalence-by-who-region-and-age-group-table.tsx:150]: So arboviruses and who regions that result in N/As across the board aren't included as visible from the filtered file I attached in the PR description.

**Review by EliasPinno — APPROVED** [review]

**Pull request merged**